### ISSUE TITLE
Create reproducible synthetic holdout evaluation harness

### ISSUE THREAD

**Issue by tomotvos** [tomotvos/ronchi-ml#6]
## Goal

Create a single reproducible evaluation command that runs a trained checkpoint against held-back synthetic Ronchi images and emits machine-readable metrics plus a short human-readable report.

## Context

Held-back synthetic images are closer to the training distribution than real images, but inference is still not as accurate as expected. We need a repeatable way to quantify that problem before changing the model or adding defects.

## Relevant files

- src/train.py — training script
- src/infer.py — inference script
- src/data.py — dataset/manifest loader
- src/eval_jsonl.py — existing eval utilities
- src/model.py — model definition
- data/manifest.jsonl, data/holdout/ — manifest and holdout data
- docs/synthetic-evaluation-plan.md

## Non-goals

- Do not retrain the model.
- Do not change the model architecture.
- Do not introduce defective mirrors.
- Do not tune hyperparameters.
- Do not optimize for real images yet.

## Proposed approach

Add or adapt an evaluation entry point that accepts:
- checkpoint path
- heldout manifest path
- image root path
- output directory
- optional limit for quick smoke tests
- optional device argument

Output directory format: runs/eval/YYYY-MM-DD-HHMM-synthetic-holdout/

Output files: metrics.json, predictions.csv, grouped_metrics.csv, config.json, summary.md

At minimum include: true p_corr, predicted p_corr, absolute p_corr error, true/predicted offset if available, f, lpi, and any dimensionless parameters used by the project.

## Acceptance criteria

- A single command runs evaluation from a clean checkout.
- The command does not require notebook-only steps.
- The command supports a small smoke-test limit.
- Results are written to a timestamped output directory.
- metrics.json includes aggregate MAE for p_corr.
- If offset labels are available, metrics.json includes offset MAE.
- predictions.csv contains one row per evaluated image.
- grouped_metrics.csv breaks errors down by relevant synthetic parameters.
- summary.md briefly explains the run and highlights worst error modes.
- Add or update docs/synthetic-evaluation-plan.md.

## Validation command or artifact

```bash
python src/eval_synthetic_holdout.py \
  --checkpoint path/to/checkpoint.pt \
  --manifest data/holdout/manifest.jsonl \
  --image-root data/holdout \
  --output-dir runs/eval/test-synthetic-holdout \
  --limit 32
```

(Update with actual script name/args for this repo.)

## Notes for future coding agent

Prefer a boring, explicit script over a clever notebook. The goal is repeatability and inspectability, not elegance.
